tsdb/head.go

```diff
@@ -428,26 +435,34 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 			Name: "prometheus_tsdb_data_replay_duration_seconds",
 			Help: "Time taken to replay the data on disk.",
 		}),
-		samplesAppended: prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name: "prometheus_tsdb_head_samples_appended_total",
-			Help: "Total number of appended samples.",
-		}, []string{"type"}),
-		outOfOrderSamplesAppended: prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name: "prometheus_tsdb_head_out_of_order_samples_appended_total",
-			Help: "Total number of appended out of order samples.",
-		}, []string{"type"}),
-		outOfBoundSamples: prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name: "prometheus_tsdb_out_of_bound_samples_total",
-			Help: "Total number of out of bound samples ingestion failed attempts with out of order support disabled.",
-		}, []string{"type"}),
-		outOfOrderSamples: prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name: "prometheus_tsdb_out_of_order_samples_total",
-			Help: "Total number of out of order samples ingestion failed attempts due to out of order being disabled.",
-		}, []string{"type"}),
-		tooOldSamples: prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name: "prometheus_tsdb_too_old_samples_total",
-			Help: "Total number of out of order samples ingestion failed attempts with out of support enabled, but sample outside of time window.",
-		}, []string{"type"}),
+		successfulSamplesAppended: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "prometheus_tsdb_head_successful_samples_appended_total",
```

Comment:
success reason is odd, for a start I would keep the old metrics and focus on failures